Extension/src/LanguageServer/extension.ts

```diff
@@ -327,11 +327,35 @@ async function checkAndApplyUpdate(updateChannel: string): Promise<void> {
                 // then the .catch call will return a resolved promise
                 // Thusly, the .catch call must also throw, as a return would simply return an unused promise
                 // instead of returning early from this function scope
+                let config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration();
+                let originalProxySupport: string = config.inspect<string>('http.proxySupport').globalValue;
+                while (true) { // Might need to try again with a different http.proxySupport setting.
```

Comment:
I don't believe this can infinite loop...I could add some extra checks in case though.